plugins/language_kdl.lua

```diff
@@ -3,82 +3,102 @@ local syntax = require "core.syntax"
 
 local identifier = "\"?[^%d%s\\/%(%){}<>;%[%]=,\"][^%s\\/%(%){}<>;%[%]=,\"]*\"?"
 
+local node_comment = { symbols = {} }
+node_comment["patterns"] = {
+  { pattern = { "{", "}" }, type = "comment", syntax = node_comment },
+  { pattern = "[^{}]+",     type = "comment" }
+}
+
+local nested_multiline = { symbols = {} }
+nested_multiline["patterns"] = {
+  { pattern = { "/%*", "%*/" }, type = "comment", syntax = nested_multiline },
+  { pattern = "[^/*]+",         type = "comment" },
+}
+
 syntax.add {
   name = "KDL",
-  files = { "%.kdl" },
+  files = { "%.kdl$" },
   space_handling = false,
   comment = "//",
   block_comment = {"/*", "*/"},
   patterns = {
     {
-		pattern = "^%s*".. identifier .."%s*",
-		type = "keyword"
-	},--
+      pattern = "^%s*".. identifier .."%s*",
+      type = "keyword"
+    },
     { pattern = "%s+",                      type = "normal"   },
     {
-		pattern = "[{;]%s*()" .. identifier .. "%s*",
-		type = {"normal", "keyword"}
-	},--
+      pattern = "[{;]%s*()" .. identifier .. "%s*",
+      type = {"normal", "keyword"}
+    },
+
     { pattern = { "r#+\"", "\"#+" },         type = "string"   },
     { pattern = { '"', '"', '\\' },          type = "string"   },
     { pattern = "[%-+]?0x[%x_]+",            type = "number"   },
     { pattern = "[%-+]?0b[01_]+",            type = "number"   },
     { pattern = "[%-+]?0o[0-7_]+",           type = "number"   },
     {
-		pattern = "[%-+]?[%d_]+%.[%d_]+e[%-+]?[%d_]+",
-		type = "number"
-	},
-	{ pattern = "[%-+]?[%d_]+%.[%d_]+",      type = "number"   },
+      pattern = "[%-+]?[%d_]+%.[%d_]+e[%-+]?[%d_]+",
+      type = "number"
+    },
+	  { pattern = "[%-+]?[%d_]+%.[%d_]+",      type = "number"   },
     { pattern = "[%-+]?[%d_]+e[%-+]?[%d_]+", type = "number"   },
     { pattern = "[%-+]?[%d_]+",              type = "number"   },
-    { pattern = "/[%-/].-\n",                type = "comment"  },
-    { pattern = {"/%*", "%*/"},              type = "comment"  },
+    { pattern = "//.*",                      type = "comment"  },
+    {
+      pattern = { "/%-.*{", "}" },
+      type = "comment", syntax = node_comment
+    },
```

Comment:
My suggestion in #477 also covered comments like this:
```kdl
mynode /-"commented" "not commented" {
  a
  b
}
```
This doesn't, right?

Reply:
Nope, only the single parameter should be commented.
In what cases did my suggestion not work?